internals/daemon/daemon_test.go

```diff
@@ -1148,3 +1148,98 @@ services:
 	c.Assert(tasks, HasLen, 1)
 	c.Check(tasks[0].Kind(), Equals, "stop")
 }
+
+func mockShutdownSyscall(f func()) (restore func()) {
+	old := shutdownSyscall
+	shutdownSyscall = f
+	return func() {
+		shutdownSyscall = old
+	}
+}
+
+func mockCheckCapSysBoot(f func() error) (restore func()) {
+	old := checkCapSysBoot
+	checkCapSysBoot = f
+	return func() {
+		checkCapSysBoot = old
+	}
+}
+
+func (s *daemonSuite) TestSyscallPosRebootDelay(c *C) {
+	wait := make(chan int)
+	defer mockCheckCapSysBoot(func() error {
+		return nil
+	})()
+	defer mockShutdownSyscall(func() {
+		wait <- 1
+	})()
+
+	period := time.Millisecond * 25
+	syscallReboot(period)
+	start := time.Now()
+	<-wait
```

Comment:
These kind of waits should probably always be in a `select` with an alternative timeout branch, so the test doesn't hang forever in case of a bug or edge case we've forgotten. Similar below. Something like this:

```go
select {
case <-wait:
case <-time.After(10 * time.Second):
	c.Fail("timed out waiting for X")
}
```

Reply:
Fixed. Thanks for the feedback.